### ISSUE TITLE
Worked example in README not working: "Error in data.frame(timestamp = Sys.time(), snapshotDate = snapshotDate,  :    arguments imply differing number of rows: 1, 0"

### ISSUE THREAD

**Issue by andrewjohnlowe** [RevolutionAnalytics/checkpoint#257]
Hello,

My end goal: I'm trying to get checkpoint to work with knitr and rmarkdown. I opened issue #216 about this a while ago, which was closed after a vignette was created with a hack to make it work, but now this is missing. Is there a new recipe? I'm unable to follow the instructions in the README.md [here](https://github.com/RevolutionAnalytics/checkpoint#using-knitr-and-rmarkdown-with-checkpoint) -- what am I missing?

So I took a step backwards, and I tried to get the [Worked example](https://github.com/RevolutionAnalytics/checkpoint#worked-example) running, but failed. I cut and paste the example into foo.R, which when sourced output the following error:

```
> source('~/Test/foo.R')
Scanning for packages used in this project
- Discovered 2 packages
Installing packages used in this project 
 - Installing ‘MASS’
MASS
Error in data.frame(timestamp = Sys.time(), snapshotDate = snapshotDate,  : 
  arguments imply differing number of rows: 1, 0
```

What am I doing wrong?

```
> sessionInfo()
R version 3.4.0 (2017-04-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] checkpoint_0.4.2

loaded via a namespace (and not attached):
[1] compiler_3.4.0 tools_3.4.0    yaml_2.1.14  
```

**Comment by andrie** [comment]
I haven't been able to find a reliable way to do this.  I thought I had a solution, so I put this into a vignette, but this caused `R CMD check` to fail, and I removed the vignette.

**Comment by andrewjohnlowe** [comment]
Yes, I gathered that from the commit messages. Currently I'm using my own functions that use install_version from devtools to install specific versions, but I don't know if the dependencies are taken care of too, and it's a bit hacky. This also worked with Bioconductor; I installed from the github read-only mirror, but I see now that it's deprecated and there is a warning to say that it will be removed. So back to square one on this issue. :-(
